homeassistant/components/mqtt/discovery.py

```diff
@@ -27,30 +27,17 @@
     'light', 'sensor', 'switch', 'lock', 'climate',
     'alarm_control_panel']
 
-ALLOWED_PLATFORMS = {
-    'binary_sensor': ['mqtt'],
-    'camera': ['mqtt'],
-    'cover': ['mqtt'],
-    'fan': ['mqtt'],
-    'light': ['mqtt', 'mqtt_json', 'mqtt_template'],
-    'lock': ['mqtt'],
-    'sensor': ['mqtt'],
-    'switch': ['mqtt'],
-    'climate': ['mqtt'],
-    'alarm_control_panel': ['mqtt'],
-}
-
-CONFIG_ENTRY_PLATFORMS = {
-    'binary_sensor': ['mqtt'],
-    'camera': ['mqtt'],
-    'cover': ['mqtt'],
-    'light': ['mqtt'],
-    'sensor': ['mqtt'],
-    'switch': ['mqtt'],
-    'climate': ['mqtt'],
-    'alarm_control_panel': ['mqtt'],
-    'fan': ['mqtt'],
-}
+CONFIG_ENTRY_COMPONENTS = [
+    'binary_sensor',
+    'camera',
+    'cover',
+    'light',
+    'sensor',
+    'switch',
+    'climate',
+    'alarm_control_panel',
+    'fan',
```

Comment:
Please don't remove ALLOWED_PLATFORMS yet, lock is still missing.

Reply:
Lock was added in commit 589764900ab432f46af60fc2b0d8471c2b2323d8, so it can be removed.

Reply:
Well then you need to rebase this PR on top of that

Reply:
Done, sorry about that!